src/modules/cpu/kernel/ricap.hpp

```diff
@@ -0,0 +1,1693 @@
+                                  rpp::Handle& handle)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for (int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi1, roi2, roi3, roi4;
+        RpptROIPtr roiPtrInput1, roiPtrInput2, roiPtrInput3, roiPtrInput4;
+        roiPtrInput1 = &roiPtrInputCropRegion[0];
+        roiPtrInput2 = &roiPtrInputCropRegion[1];
+        roiPtrInput3 = &roiPtrInputCropRegion[2];
+        roiPtrInput4 = &roiPtrInputCropRegion[3];
+        compute_roi_validation_host(roiPtrInput1, &roi1, &roiDefault, roiType);
+        compute_roi_validation_host(roiPtrInput2, &roi2, &roiDefault, roiType);
+        compute_roi_validation_host(roiPtrInput3, &roi3, &roiDefault, roiType);
+        compute_roi_validation_host(roiPtrInput4, &roi4, &roiDefault, roiType);
+
+        Rpp8u *srcPtrImage1, *srcPtrImage2, *srcPtrImage3, *srcPtrImage4, *dstPtrImage;
+        int permutedCount = batchCount * 4;
+        srcPtrImage1 = srcPtr + (permutedIndices[permutedCount] * srcDescPtr->strides.nStride);
+        srcPtrImage2 = srcPtr + (permutedIndices[permutedCount + 1] * srcDescPtr->strides.nStride);
+        srcPtrImage3 = srcPtr + (permutedIndices[permutedCount + 2] * srcDescPtr->strides.nStride);
+        srcPtrImage4 = srcPtr + (permutedIndices[permutedCount + 3] * srcDescPtr->strides.nStride);
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength1 = roi1.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength2 = roi2.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength3 = roi3.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp32u bufferLength4 = roi4.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp8u *srcPtrChannel1, *srcPtrChannel2, *srcPtrChannel3, *srcPtrChannel4, *dstPtrChannel;
+        srcPtrChannel1 = srcPtrImage1 + (roi1.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi1.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel2 = srcPtrImage2 + (roi2.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi2.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel3 = srcPtrImage3 + (roi3.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi3.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        srcPtrChannel4 = srcPtrImage4 + (roi4.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi4.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength1 = (bufferLength1 / 48) * 48;
+        Rpp32u alignedLength2 = (bufferLength2 / 48) * 48;
+        Rpp32u alignedLength3 = (bufferLength3 / 48) * 48;
+        Rpp32u alignedLength4 = (bufferLength4 / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
```

Comment:
For L56-L92, using arrays for pointers may be much cleaner
```
RpptROI roi[4];
RpptROIPtr roiPtrInput[4];

for (int i = 0; i < 4; i++)
{
    roiPtrInput[i] = &roiPtrInputCropRegion[i];
    compute_roi_validation_host(&roiPtrInput[i], &roi[i], &roiDefault, roiType);
    srcPtrImage[i] = srcPtr + (permutedIndices[permutedCount + i] * srcDescPtr->strides.nStride);
    bufferLength[i] = roi[i].xywhROI.roiWidth * layoutParams.bufferMultiplier;
    srcPtrChannel[i] = srcPtrImage[i] + (roi[i].xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi[i].xywhROI.xy.x * layoutParams.bufferMultiplier);
    alignedLength[i] = (bufferLength[i] / 48) * 48;
}
```
Please re-evaluate and make all instances of this concise.

Reply:
Sure